lib/still/preprocessor/html_responsive_image.ex

```diff
@@ -0,0 +1,93 @@
+defmodule Still.Preprocessor.HtmlResponsiveImage do
+  @no_responsive_image "no-responsive-image"
+
+  @moduledoc """
+  Parses an HTML `Still.SourceFile` and replaces HTML img with responsive images.
+  Images that have the attributes "srcset" or "#{@no_responsive_image}" are ignored.
+  """
+
+  alias Still.{SourceFile, Preprocessor}
+  alias Still.Compiler.TemplateHelpers.ResponsiveImage
+
+  use Preprocessor
+
+  import Still.Utils
+
+  def render(
+        %SourceFile{
+          extension: ".html",
+          content: content,
+          input_file: input_file
+        } = source_file
+      ) do
+    {:ok, document} = Floki.parse_document(content)
+
+    new_content =
+      document
+      |> Floki.find_and_update("img", fn
+        {"img", img_attrs} ->
+          src = find_node_attr(img_attrs, "src")
+          srcset = find_node_attr(img_attrs, "srcset")
+          no_responsive_image = find_node_attr(img_attrs, @no_responsive_image)
+
+          if is_image?(src) && is_nil(srcset) && is_nil(no_responsive_image) do
```

Comment:
Code style: I'd abstract these `if` branches to something more meaningful. At least when doing code reviews, it's a lot harder to reason when interpreting specific conditions in a high-level abstraction function.

Reply:
Updated with a different approach!